### DISCUSSION TITLE
SamplingService and IgnoredTracerContext

### DISCUSSION THREAD

**Discussion by meteorice** [apache/skywalking#10866] [Q&A]
I'm currently using SkyWalking for application performance monitoring. However, I've encountered an issue where the IgnoredTracerContext is being used as the context for ignoring sampling within the SamplingService. While this seems to work as intended for most cases, it appears that error logs might be getting lost during exceptions.

I believe this is happening because when an exception occurs, the trace context switches to IgnoredTracerContext, potentially causing the associated error logs to be missed. I'm trying to understand if this is an inherent design decision or if there's a way to ensure that error logs are not omitted, even when the trace context is being ignored.

**Comment by wu-sheng** [comment]
I don't follow what do you mean `when an exception occurs`? Which exception?

**Reply by meteorice** [reply]
When the service logic is abnormal.
